packages/ds-ext/src/DiagramEditorProvider.ts

```diff
@@ -38,15 +39,28 @@ export class DiagramEditorProvider implements vscode.CustomEditorProvider<Diagra
   }
 
   dispose(): void {
-    this.duckDBStorage.close();
+    if (this.duckDBStorage instanceof DuckDBStorage) {
+      this.duckDBStorage.close();
```

Comment:
Consider updating ObserverStorage with an optional method `dispose`. Then DiagramEditorProvider can delegate this task to the storage implementation (if it is provided).

Reply:
`dispose()` is sync but `duckDBStorage.close()` is async. Maybe it is not a problem, but can/should we await it?